python/pyrogue/_RunControl.py

```diff
@@ -140,10 +140,12 @@ def _run(self):
         #print("Thread start")
         self.runCount.set(0)
 
-        while (self.runState.valueDisp() == 'Running'):
-            time.sleep(1.0 / float(self.runRate.value()))
-            if self._cmd is not None:
-                self._cmd()
+        with self.root.updateGroup(period=1.0):
 
-            with self.runCount.lock:
-                self.runCount.set(self.runCount.value() + 1,write=False)
+            while (self.runState.valueDisp() == 'Running'):
+                time.sleep(1.0 / float(self.runRate.value()))
```

Comment:
@slacrherbst Should `time.sleep(1.0 / float(self.runRate.value()))` be `time.sleep(period / float(self.runRate.value()))` (use period arg instead of hard coded 1.0)?

Reply:
No run rate is in hz, so the sleep period is 1/rate.